Add Early Lunar Habitation

WithCrow opened the pull request:
Taking over from Mmeridian - the Early Lunar Habitation Program takes inspiration from the Apollo Applications Program in how it is structured. It continues Lunar exploration right after the Crewed Lunar Landing program finishes and allows the players to continue using their Lunar LVs / evolve them from there. 

The contracts are, in order:

- Two-Week Crewed Lunar Stay 
- 1-Month Crewed Lunar Stay
- 3-Month Crewed Lunar Stay
- 6-Month Crewed Lunar Stay

With the optional:

- Lunar Orbit Space Station
- 3-Month Crewed Lunar Orbit Stay
- 6-Month Crewed Lunar Orbit Stay
- 12-Month Crewed Lunar Orbit Stay

After talking to @CXG-2827, the following bulletpoints need to be evaluated:

- Extended Lunar Orbital Rendezvous (ELOR) contract (optional/mandatory?) in order to prepare players for Mars
- As this PR also adds science experiments (taking inspiration from the Early Space Stations program), the specific science values might have to be tweaked (both for ESS & ELH). 
- The funding is most likely not balanced at the moment, as one can complete the entire program for a fraction of what it funds. The "mean" approach is something that would have to be determined, so that the funding can be tailored for that.

Mmeridian committed:
Updates I forgot to push

Mmeridian committed:
Catching up/cleaning up old work

Mmeridian committed:
Fixes to get contracts to load, Kerbalism config window support

Mmeridian committed:
Merge pull request #2 from KSP-RO/master

Update

Mmeridian committed:
Fix compat w/ stations program?

WithCrow committed:
Initial Fix + Moving MK2 Lander Cab

WithCrow committed:
Lunar Station Content Addition

Adds more content to Lunar Stations

WithCrow committed:
More Content

WithCrow committed:
Added LunarOrbit experiments in Configure.cfg

WithCrow committed:
Rewrote the long duration lunar orbital habitation contract

Mmeridian committed:
Merge pull request #3 from Mmeridian/science-fix

Intial Fixes + Lunar Station Content + MK2 Cab moved

WithCrow committed:
Updating LOS

WithCrow committed:
Merge branch 'master' of https://github.com/Mmeridian/RP-1_Mm

WithCrow committed:
Added installation guide

WithCrow committed:
Update Installation.md

WithCrow committed:
Updated installation guide

WithCrow committed:
Changed name (& tags) to Early Lunar Habitation

This brings it in line with the 'Early Space Station' program and allows future programs to use the Lunar Habitation name.

WithCrow committed:
Long Duration Station Contracts

WithCrow committed:
Small fix

WithCrow committed:
Update Installation.md

WithCrow committed:
Fix

WithCrow committed:
Names fixed

WithCrow committed:
More fixes

I noticed a lot of the contracts had incorrect names for contracts. The worst offenders were my own X Month Lunar Orbital Stay missions, but I noticed the MoonSpaceStation contract was somewhat strange, in that its name was moon_SpaceStation and not MoonSpaceStation (keeping in line with other contracts of ELH), so I've fixed that

WithCrow committed:
Installation.md is obsolete & hence deleted

siimav commented on GameData/RP-1/Programs/Programs.cfg:
```diff
@@ -1926,39 +1926,69 @@ RP0_PROGRAM
+		MoonSpaceStation = true
+		MoonSpaceStation3Month = true
+		MoonSpaceStation6Month = true
+		MoonSpaceStatio123Month = true
```
Typo

siimav replied:
This is still wrong, btw.

siimav commented on GameData/RP-1/Contracts/Early Lunar Habitation/MoonSpaceStation.cfg:
```diff
@@ -0,0 +1,139 @@
+			resource = ResearchPayload
+			minQuantity = 700
+		}
+        PARAMETER
```
A bunch of indentation issues in various files.

siimav commented on GameData/RP-1/Tree/TREE-Parts.cfg:
```diff
@@ -26163,7 +26163,7 @@
 {
-    %TechRequired = lunarLanding
+    %TechRequired = advancedUncrewedLanding
     %cost = 4500
```
Are these accidental changes? If not then these need to be accompanied with part browser json changes.

WithCrow replied:
This was done on purpose, as the only part that would fulfill the requirements for the 2 week stay was the LM Shelter part - I moved the mk2 cab so that non-Apollo hardware could be used. 

I'll get on it, but I do have to ask - why? It worked fine in my game, so I'm curious what the part browser json would affect beyond this (but again, I will change it)

siimav reviewed (commented):
Do I understand correctly that the player needs to launch 3 separate bases where 2 of them need to target a randomly generated spot on the Moon? I would personally prefer if a single location is allowed and just force the player to do crew rotations of various lengths.

WithCrow commented:
> Do I understand correctly that the player needs to launch 3 separate bases where 2 of them need to target a randomly generated spot on the Moon? I would personally prefer if a single location is allowed and just force the player to do crew rotations of various lengths.

The randomly generated points are @Mmeridian's work - I believe the intention was not to require the players to land in a specific location, but to to visit points generated _after_ landing in a rover/etc. I'll speak to him more on this, but I'm aware that the current implementation does not work the way it should. 

I haven't had the time to review those particular contracts & fix them, but it is on the To-Do. 

In regards to the aim of the program: this is something that @CXG-2827 and I have talked about & haven't settled on. The way it's set up at the moment is that the player launches a early outpost based on their Crewed Lunar Landing hardware and then launches a dedicated base down the line. 

The player would, in the current implementation, need to launch at least two generations of bases (/base modules), as the final science experiment is in the Modular Stations tech node. The program can be simplified by removing that last tier of surface experiments & stretching out the time frame of the first one - by checking for completion of the experiment, like the Lunar Orbital contracts do - I am curious to hear what others think.

njits23 commented on GameData/RP-1/Science/Configure.cfg:
```diff
@@ -1072,6 +1072,34 @@
+			  type = Experiment
+			  id_field = experiment_id
+			  id_value = RP0lunarHabit2
+			}
```
This is also incorrectly indented.

njits23 commented on GameData/RP-1/Science/Configure.cfg:
```diff
@@ -1275,6 +1303,104 @@
+			{
+			  type = Experiment
+			  id_field = experiment_id
+			  id_value = RP0LunarOrbitHabit1
```
There's a bunch more like this, just search for tabs vs spaces.

Mmeridian commented:
Crow is right that the intention was to create some waypoints near your own chosen landing site that would require you to drive to them similar to the targeted landing contracts. the stretch goal was to require surface samples be performed and then analyzed at a lab (whether you drive your lab or just bring the samples back to a lab is up to you) unfortunately that functionality was never fully built out before I told Crow to take over for me 🤕

On the note of ISRU, don't personally think it should be mandatory but adding experiments tied to ISRU could be really cool. But I also don't expect to contribute in the near future so whatever yall decide is fine with me lol

WithCrow commented on GameData/RP-1/Contracts/Early Lunar Habitation/MoonSpaceStation12Month.cfg:
```diff
@@ -0,0 +1,64 @@
+	// ************ PARAMETERS ************
+	
+	PARAMETER
+		{
```
Thank you for the effort of pointing them out!

WithCrow committed:
Clean-Up & Indentation Fix

I hope this clears up the indentation issue, I've also removed the waypoint feature

CXG-2827 committed:
typo and tweaks

CXG-2827 commented:
FYI, I added a requirement for the Early Station program to be either active or completed to allow taking the Early Lunar habitation program.

Something worth discussing is whether the 30-day lunar outposts or Lunar Station contracts should be locked behind the first (Earth) station contract. From reading the following AAP document https://ntrs.nasa.gov/citations/19660022516 pages 53-54, it looks like long duration lunar orbits and lunar surface activities would be performed after long-duration earth orbit missions.

I figure an extended 14D lunar landing can be left as-is, bur for any longer on the surface or in orbit, that has to be proven out in LEO first.

And a question I have is whether these duration contracts should also include a duration parameter rather than just relying on the transmission of the science. You technically can deploy 2 stations in parallel and complete the contract in less time since both stations can be transmitting simultaneously. Though a drawback of the duration parameter is that it can reset if you go on EVA and no longer meet the MinCrew parameter, or if the outpost pops off the surface after a timewarp and no longer is in a Landed state.

WithCrow committed:
Added LunarSurveyor contract

WithCrow committed:
Fix - Forgot to add it to Programs.cfg

CXG-2827 committed:
Gating fixes and tweaks

siimav commented on GameData/RP-1/Contracts/Early Lunar Habitation/MoonLandingCrew6Month.cfg:
```diff
@@ -0,0 +1,117 @@
+			}
+			PARAMETER
+			{
+			    name = researchPayloadCapacity //make sure you can hold at least 90 days of research payload
```
Spaces instead of a single tab

WithCrow commented on GameData/RP-1/Contracts/Early Lunar Habitation/LunarSurveyor.cfg:
```diff
@@ -0,0 +1,107 @@
+				completionText = Stable orbit: Confirmed
+			}
+		}
+		}
```
Yep, shouldn't be there - removed it.

WithCrow replied:
Spot on - pushed now

WithCrow committed:
Format checking & config json

WithCrow committed:
More format fixes

It would appear the same format error where spaces had been used instead of tab appears in the other crewed landing contracts as well.

WithCrow committed:
Removed bracket

CXG-2827 committed:
Add Bracket

WithCrow committed:
Amend StationExperiments to resolve conflict

WithCrow commented on GameData/RP-1/Programs/Programs.cfg:
```diff
@@ -1926,39 +1926,79 @@ RP0_PROGRAM
 		}
+		ANY
+		{
+			active_program = EarlyEarthSpaceStation
+			complete_program = EarlyEarthSpaceStation
```
`	REQUIREMENTS
	{
		complete_program = CrewedLunarANY
		ANY
		{
			active_program = EarlyEarthSpaceStation
			complete_program = EarlyEarthSpaceStation
		}
	}`
	
Something like this, right?

siimav commented on GameData/RP-1/Programs/Programs.cfg:
```diff
@@ -1926,39 +1926,79 @@ RP0_PROGRAM
-
 	OBJECTIVES
 	{
-		complete_contract = TODO
```
Might as well use the shorter syntax

WithCrow replied:
Please elaborate on what you mean?

CXG-2827 commented on GameData/RP-1/Programs/Programs.cfg:
```diff
@@ -1926,39 +1926,79 @@ RP0_PROGRAM
+		MoonSpaceStation3Month = true
+		MoonSpaceStation6Month = true
+		MoonSpaceStation12Month = true
+		MoonOrbitCrewedPolar = true
+		MoonLandingCrew1Year = true
```
I have a crewed polar contract from the closed PR I can add

WithCrow committed:
Experiment restructuring & 1 Year optional

WithCrow committed:
Revert on StationExperiments format & 6 Month crew change

WithCrow committed:
Resolve conflict (again)

WithCrow committed:
Actually resolve conflict

I'm getting way too tired

WithCrow commented:
Credit for the experiment restructuring to @njits23, who fixed the experiments! I've added the 1 year optional surface stay, as well as changed the 6 month crew requirement to just 3, as this tightens the program's scope and makes the progression smoother, compared to the sudden jump in difficulty / logistics required due to the previous 6 crew requirement.

siimav commented:
🤖found syntax errors in files

njits23 commented on GameData/RP-1/Science/Experiments/CrewScience/SurfaceExperiments.cfg:
```diff
@@ -0,0 +1,187 @@
+// Month-Long Lunar Habitation
+// ============================================================================
+EXPERIMENT_DEFINITION
+
```
Opening curly bracket is missing.

CXG-2827 committed:
Syntax Scientists and Polar Orbit

WithCrow committed:
Unified ID scheme

Capkirk123 commented:
resolves https://github.com/KSP-RO/RP-1/issues/1613 ?

WithCrow commented:
I believe so, yeah. Unless they're still active? The PR doesn't seem to suggest so, though

WithCrow committed:
Resolve conflict with StationExperiments

Had I known my indentation fix PR would be merged that quickly, I could have just left things as is. Oh well

WithCrow committed:
Resolve attempt 2

WithCrow committed:
Resolve pt3

WithCrow commented:
Okay, I can't really tell why it's marking things as a conflict now, while it didn't before. If someone could take a look at it, I would greatly appreciate it

siimav commented:
Merge [KSP-RO:master](https://github.com/KSP-RO/RP-1/tree/master) into your master branch.

WithCrow committed:
Merge branch 'master' into update_rp4

WithCrow committed:
Merge pull request #5 from Mmeridian/update_rp4

Update rp4

WithCrow committed:
Fix incorrect update

WithCrow committed:
Remove sample masses

WithCrow commented on GameData/RP-1/Science/Experiments/CrewScience/SurfaceExperiments.cfg:
```diff
@@ -0,0 +1,188 @@
+	title = Extended Lunar Habitation with Analysis
+	baseValue = 600 
+	scienceCap = 600
+	dataScale = 128 //generates 10 samples over the 360d experiment
```
I must have mixed things up when squashing the longer duration experiments into this one. Fixed, thanks for pointing it out.

WithCrow committed:
Correct data rate for 2 Year Surface Experiment

WithCrow commented:
As much as it pains me to say, I don't think so. Due to hardware issues on my end, development has gone stale and I still think more playtesting should be done. As far as I know, no one has played the entire program in its current iteration. Njits suggested adding more experiments, but I've not had much luck finding appropriate experiments for the era - if anyone has good ideas, suggestions would be greatly appreciated.

WithCrow committed:
Merge branch 'master' into update_rp4_2

WithCrow committed:
Update to RP4 1Oct25

Update 1Oct25

WithCrow committed:
Align with new lunar landing experiment assignment format

WithCrow commented:
After njits gave the clear [here](https://discord.com/channels/319857228905447436/1115321536807718992/1423029975279341609) that additional experiments are more in the 'nice to have, but not mandatory' category and combined with me being unable to find a lot of sources for appropriate lunar base experiments, the only thing standing in the way of this being merged is more data / feedback from people playtesting it.

Karhgath commented:
I was even thinking about a scenario where you would take Early Station before Moonboots. We could gate some habs contract (e.g. polar crew orbit, moon space station) on Lunar Explo contracts, you could start putting a station in even before landing crew on the surface (initial Moon missions architecture were not even about landing, so early moon station before landing could make althist sense).

Just trying to add variety and paths through the programs.

Lunar habs is well paying but not close to lunar explo/stations and a slow ramp up so I could see them all working in parallel.

CXG-2827 committed:
Adjusted confidence and contract agents

WithCrow committed:
Ad experiments to later nodes

WithCrow committed:
Merge branch 'master' of https://github.com/Mmeridian/RP-1_Mm

Karhgath committed:
Rearranged and simplified lunar habs experiments

Using same selector slot now, newer part have access to older parts experiments as well, and fixed a typo in the configs.

CXG-2827 committed:
Flat Funding Curve

Karhgath committed:
Merge branch 'master' into fix-crew-science

WithCrow commented:
Personally, I consider it done. I'm aware of two people having played through it without problems & more are currently playing with it.RiS 25 is also playing with it, but as far as I know, no one's close to lunar habitation currently.

WithCrow commented:
The 3 Month Surface stay (incorrectly) lists requiring 750 Research Payload, when it only requires 500. I will change this after RiS, as I hope to receive more feedback after.

On adifferent note, I am debating removing the 'Return Home Safely' requirement from the surface stays - which would allow completing them in one go. Currently, you _need_ to do multiple crew rotations and I would rather make them optional. The Stations program also doesn't require crew rotations, so I don't see much value in enforcing it here.

CXG-2827 committed:
Various description fixes payload corrections and return home removals

CXG-2827 committed:
Program gating and other minor tweaks

WithCrow committed:
Merge pull request #8 from KSP-RO/master

Sync branches

CXG-2827 committed:
Caught a few more

CXG-2827 committed:
Fixed Crewed Polar Orbit Contract

CXG-2827 committed:
Lunar Surveyor Gating

WithCrow committed:
1 Month Surface now in ESS

WithCrow commented:
@jimmymcgoochie pointed out that currently, the 1 month experiment is unlocked a material science node before the 2 week surface contract. Due to this, I've moved the 1 month experiment back two nodes into Early Space Stations, as opposed to its previous assignment in prototype space stations.

CXG-2827 committed:
Lunar Surveyor Tweak

CXG-2827 committed:
Fixes and Tweaks

siimav commented:
The Month-Long Lunar Habitation experiment shouldn't come pre-included with every freshly picked part. Perhaps it shouldn't be a separate button in the UI either.

siimav commented on GameData/RP-1/Science/Configure.cfg:
```diff
@@ -1367,6 +1367,173 @@
+			name = Month-Long Lunar Habitation
+			desc = In-situ health analysis experiments performed by astronauts while working on the Moon. <b>NOTE: This experiment takes 30 days and consumes 700L of Research Payload</b> //Placeholder Text.
+			mass = 0.01 
+			tech = spaceStationPrototypes
```
This is a bit odd that the experiments show up in the first station parts node.

CXG-2827 commented:
What about "Short-Term Lunar Habitation" for the 14 day experiment that aligns with the LEM shelter?

Regarding the orbital experiment, I wonder if the "multi-week Lunar Orbit Habitation" experiment should just be dropped and modify the contract to just work with the Multi-Month experiment. Comparing it to the stations program, Multi-Week takes 180 days for the LEO version while Early lunar has it just for 14 days.

WithCrow committed:
Correctly assign 1 Month experiment

WithCrow committed:
Simplify & rename experiments

WithCrow commented:
Upon thinking about it, I've also removed the 1 month surface experiment. The reason for this being that it doesn't really introduce anything new, now that both it and the 2 year experiment are assigned in Early Space Stations. The two weeks experiment has been renamed to Short-Term Lunar Habitation, the two year one to Long-Term Lunar Habitation.

This also means that we add three experiments instead of four:
- Two week surface experiment (LM Shelter, etc.)
- (Up to) 2 year lunar surface experiment 
- (up to) 2 year lunar orbit experiment

WithCrow committed:
Tweak contract parameter description

WithCrow commented:
Very good point, has been done

WithCrow committed:
Merge pull request #9 from KSP-RO/master

Bring up to date

CXG-2827 committed:
3 Crew and fraction tweak

CXG-2827 committed:
one month description

CXG-2827 committed:
more edits

WithCrow committed:
Account for science modifiers

CXG-2827 committed:
Nerf Crewed Polar Orbit

siimav commented:
The fact that there's exactly 0 landed crew science experiments other than the 2 required ones feels quite disappointing TBH. Especially compared to the Early Stations program. Not sure if there is anything interesting out there in IRL proposals.

WithCrow committed:
Add more experiments

Add more lunar surface experiments

WithCrow committed:
Move SSPX base parts, remove redundant experiment module

WithCrow commented:
I've moved the airlock and some (2.25 & 4.5m) base frames up to ESS

siimav commented:
Some of the new experiments appear to be broken.
<img width="440" height="151" alt="image" src="https://github.com/user-attachments/assets/435e62ac-b828-4d1c-a48d-352d4d72b6d1" />
<img width="438" height="164" alt="image" src="https://github.com/user-attachments/assets/ff5c351f-46cf-4a38-86aa-9096ef5dd53f" />
<img width="438" height="142" alt="image" src="https://github.com/user-attachments/assets/32d5eb29-ac9e-48ba-9e33-f5552b5c14eb" />
<img width="433" height="154" alt="image" src="https://github.com/user-attachments/assets/fe570880-0ac2-42a5-bdb1-e2d10d079f60" />

WithCrow committed:
Fix experiments

WithCrow commented:
It works a-okay for the proto tier, but for the early tier, there are some weird things happening. Long duration ALSEP shows up _always_ even when not selected, but lunar night astronomy only shows up if you select long duration ALSEP _again_?
<img width="1126" height="1019" alt="image" src="https://github.com/user-attachments/assets/e7055a66-06ea-43fc-aaff-8faad6c3c1ff" />
I
<img width="607" height="454" alt="image" src="https://github.com/user-attachments/assets/df5ca163-941f-4375-b6ee-f83c07ce0de9" />

<img width="585" height="453" alt="image" src="https://github.com/user-attachments/assets/61005f55-bc4e-41d6-84f0-d914dc4a51de" />

Something must be getting assigned incorrectly, I have not been able to find it. Will look at it again in a few hours.

WithCrow committed:
Experiment tweaks

siimav commented:
I'm testing on the 3.38m hab module (long) and there still appears to be a mismatch, including one experiment being always present.
<img width="808" height="302" alt="image" src="https://github.com/user-attachments/assets/c8255cc2-845b-41f1-8157-fee4c89714dd" />

<img width="801" height="282" alt="image" src="https://github.com/user-attachments/assets/b2a7cecc-3deb-4ac6-b6a1-925be43166e5" />


<img width="812" height="302" alt="image" src="https://github.com/user-attachments/assets/20560135-9b0e-4e38-831d-568638aff4be" />

WithCrow committed:
Tweaks to experiment names & funding

WithCrow committed:
Lunar surveyor is kil

WithCrow committed:
Missing bracket

WithCrow committed:
Typos

siimav commented on GameData/RP-1/Programs/Programs.cfg:
```diff
@@ -1602,6 +1591,7 @@ RP0_PROGRAM
 	
 	OPTIONALS
 	{
+		LunarSurveyor = true
```
This should get yeeted as well

WithCrow committed:
Remove Surveyor optional

CXG-2827 committed:
fraction complete tweaks

CXG-2827 committed:
science corrections

WithCrow commented:
<img width="2558" height="938" alt="image" src="https://github.com/user-attachments/assets/11e31a9e-e017-4ad6-a648-fce9ffb7f845" />
Strange. It ticked to complete without having transmitted everything, but not after all of the science had been collected

<img width="313" height="224" alt="image" src="https://github.com/user-attachments/assets/96f7cdf3-af94-4e0b-bea2-7abe4f93ba99" />
Ran the test again and...??? 

Ran it again and it also 'completed' at a different date (29 days on the moon, as opposed to the 35 in the first), but also transmitted more science than possible!
<img width="1023" height="473" alt="image" src="https://github.com/user-attachments/assets/e2798eba-df49-4ace-90e4-0710c135d25d" />

CXG-2827 committed:
Lunar surface fixes

CXG-2827 committed:
fraction completes for station

WithCrow committed:
Bring up to date

Bring up to date

WithCrow committed:
Remove outdated description

WithCrow committed:
Merge pull request #11 from KSP-RO/master

Bring up to date
